src/core/mergefilters.c

```diff
@@ -455,24 +462,27 @@ static const VSFrameRef *VS_CC maskedMergeGetFrame(int n, int activationReason,
 
 #ifdef VS_TARGET_CPU_X86
-                if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
-                    func = d->premultiplied ? vs_mask_merge_premul_byte_sse2 : vs_mask_merge_byte_sse2;
-                else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
-                    func = d->premultiplied ? vs_mask_merge_premul_word_sse2 : vs_mask_merge_word_sse2;
-                else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
-                    func = d->premultiplied ? vs_mask_merge_premul_float_sse2 : vs_mask_merge_float_sse2;
-                else
-                    continue;
-#else
-                if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
-                    func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_byte_c : vs_mask_merge_premul_byte_c) : vs_mask_merge_byte_c;
-                else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
-                    func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_word_c : vs_mask_merge_premul_word_c) : vs_mask_merge_word_c;
-                else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
-                    func = d->premultiplied ? vs_mask_merge_premul_float_c : vs_mask_merge_float_c;
-                else
-                    continue;
+                if (d->cpulevel >= VS_CPU_LEVEL_SSE2) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = d->premultiplied ? vs_mask_merge_premul_byte_sse2 : vs_mask_merge_byte_sse2;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = d->premultiplied ? vs_mask_merge_premul_word_sse2 : vs_mask_merge_word_sse2;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = d->premultiplied ? vs_mask_merge_premul_float_sse2 : vs_mask_merge_float_sse2;
+                }
 #endif
+                if (!func) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_byte_c : vs_mask_merge_premul_byte_c) : vs_mask_merge_byte_c;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_word_c : vs_mask_merge_premul_word_c) : vs_mask_merge_word_c;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = d->premultiplied ? vs_mask_merge_premul_float_c : vs_mask_merge_float_c;
+                }
+
+                if (!func)
```

Comment:
This redundancy could be avoided by moving the `continue` as `else` condition into the block above.